backend/headless/backend.c

```diff
@@ -0,0 +1,166 @@
+#include <stdlib.h>
+#include <EGL/egl.h>
+#include <EGL/eglext.h>
+#include <wlr/util/log.h>
+#include "backend/headless.h"
+#include "glapi.h"
+
+static bool backend_start(struct wlr_backend *wlr_backend) {
+	struct wlr_headless_backend *backend =
+		(struct wlr_headless_backend *)wlr_backend;
+	wlr_log(L_INFO, "Initializating headless backend");
+
+	// TODO
```

Comment:
?

Reply:
This `i < 1` and `1280, 720` need to be replaced. Maybe I'll just refuse to create outputs before backend startup, and do not create any output by default.

Reply:
Should probably cache output create requests like Wayland does I suppose.

Reply:
An issue with this is that NULL will be returned by `wlr_headless_add_output` instead of a `wlr_output`.

Reply:
Nah. It's not like backend_start is really doing anything here. You already have an EGL context ready once init is called.

Reply:
Okay, so let's make `wlr_headless_add_output` create outputs regardless of whether or not the backend is started, and start output timers on backend startup.

Should I use one timer for all outputs, or one timer per output like it's done currently?

Reply:
One timer per output and make the framerate configurable. Let's also make the resolution reconfigurable at runtime.